ouroboros-consensus/test/storage-test/Test/Ouroboros/Storage/ChainDB/Model.hs

```diff
@@ -873,12 +772,8 @@ validChains cfg m bs =
     -- first, which results in the valid chain A->B', which is then chosen
     -- over the equally preferable A->B as it will be the first in the list
     -- after a stable sort.
-    sortChains $ pruneKnownInvalid <$> chains bs
+    sortChains $ chains bs
```

Comment:
Thecomment is a little bit confusing, because it sounds like it suggests that sorting before validating ensures that the model and SUT pick the same chain. However, knowing about valid/invalid blocks in advance doesn't (or shouldn't) change the outcome of chain selection. I think what the comment is saying is that we have to sort the chains (before or after validating, doesn't really matter) such that the model and SUT pick the same chain.

Reply:
I agree that this comment is confusing; in particular, it seems very unlikely that we ever have equally preferable chains as we use (non-cryptographic, but still) hashes as tie breakers. Maybe this 

In any case, I don't think we have to change this in this PR (this comment is much older than any of the PRs that are reverted here, in particular, it was not changed when `pruneKnownInvalid` was introduced), but I think clarifying this is a useful separate change.

Reply:
Opened https://github.com/input-output-hk/ouroboros-consensus/issues/298